File: frontend/src/types/files.ts
export interface FileRecord {
  id: string;
  original_name: string;
  file_type: "spreadsheet" | "document";
  mime_type: string;
  file_size: number;
  table_name: string | null;
  row_count: number | null;
  column_names: string[] | null;
  status: string;
  error_message: string | null;
  created_at: string;
  updated_at: string;
}

export interface FileListResponse {
  files: FileRecord[];
  total: number;
}

File: backend/app/schemas/files.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class FileResponse(BaseModel):
    id: uuid.UUID
    original_name: str
    file_type: str
    mime_type: str
    file_size: int
    table_name: str | None = None
    row_count: int | None = None
    column_names: list[str] | None = None
    status: str
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class FileListResponse(BaseModel):
    files: list[FileResponse]
    total: int
